tests/TestCase/ExceptionsTest.php

```diff
@@ -0,0 +1,185 @@
+
+        $exception = new $class('message', null, $previous);
+        $this->assertSame('message', $exception->getMessage());
+        $this->assertSame($defaultCode, $exception->getCode());
+        $this->assertSame($previous, $exception->getPrevious());
+    }
+
+    /**
+     * Tests FatalErrorException works.
+     *
+     * @return void
+     */
+    public function testFatalErrorException()
+    {
+        $previous = new Exception();
+
+        $exception = new FatalErrorException('message', 100, __FILE__, 1, $previous);
+        $this->assertSame('message', $exception->getMessage());
+        $this->assertSame(100, $exception->getCode());
+        $this->assertSame(__FILE__, $exception->getFile());
+        $this->assertSame(1, $exception->getLine());
+        $this->assertSame($previous, $exception->getPrevious());
+
+        $exception = new FatalErrorException('message', null, __FILE__, 1, $previous);
+        $this->assertSame('message', $exception->getMessage());
+        $this->assertSame(500, $exception->getCode());
+        $this->assertSame(__FILE__, $exception->getFile());
+        $this->assertSame(1, $exception->getLine());
+        $this->assertSame($previous, $exception->getPrevious());
+    }
+
+    /**
+     * Tests PHP7ErrorException works.
+     *
+     * @return void
+     */
+    public function testPHP7ErrorException()
+    {
+        $this->skipIf(version_compare(PHP_VERSION, '7.0.0', '<'));
+
+        $previous = new Exception();
+        $error = new Error('message', 100, $previous);
+        $line = __LINE__ - 1;
+
+        $exception = new PHP7ErrorException($error);
+        $this->assertSame(100, $exception->getCode());
+        $this->assertSame(__FILE__, $exception->getFile());
+        $this->assertSame($line, $exception->getLine());
+        $this->assertSame($previous, $exception->getPrevious());
+    }
+
+    /**
+     * Tests PersistenceFailedException works.
+     *
+     * @return void
+     */
+    public function testPersistenceFailedException()
+    {
+        $previous = new Exception();
+        $entity = new Entity();
+
+        $exception = new PersistenceFailedException($entity, 'message', 100, $previous);
+        $this->assertSame('message', $exception->getMessage());
+        $this->assertSame(100, $exception->getCode());
+        $this->assertSame($previous, $exception->getPrevious());
+        $this->assertSame($entity, $exception->getEntity());
+
+        $exception = new PersistenceFailedException(new Entity, 'message', null, $previous);
+        $this->assertSame('message', $exception->getMessage());
+        $this->assertSame(500, $exception->getCode());
+        $this->assertSame($previous, $exception->getPrevious());
+    }
+
+    /**
+     * Provides pairs of exception name and default code.
+     *
+     * @return array
+     */
+    public function exceptionProvider()
+    {
+        return [
+            ['Cake\Console\Exception\ConsoleException', 500],
```

Comment:
It could. But then, test cases will pass even if unexpected code is defined in the class.